EXP-Client/progress/TaskItem.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import { View, Text, TextInput, TouchableOpacity, Button, StyleSheet } from 'react-native';
+import { IconButton } from 'react-native-paper';
+import { Rating } from 'react-native-ratings';
+
+export default function TaskItem({ task }) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [name, setName] = useState(task.name);
+  const [deadline, setDeadline] = useState(task.deadline);
+  const [isCompleted, setIsCompleted] = useState(task.completed);
+  const [rating, setRating] = useState(task.rating);
+
+  const onDelete = (name) => {
+    console.log(`Deleted ${name}`)
+  };
+
+  const onUpdate = (name) => {
+    console.log(`Updated ${name}`)
+  };
+
+  const handleSave = () => {
+    onUpdate({
+      ...task,
+      name,
+      deadline,
+      isCompleted,
+      rating,
+    });
+    setIsEditing(false);
+  };
+
+  return (
+    <View style={styles.container}>
+      {!isEditing ? (
+        <>
+          <View style={{flexDirection: 'row'}}>
+            <View style={styles.content}>
+              <Text style={styles.taskName}>{task.name}</Text>
+              <Text style={styles.taskDeadline}>{task.deadline}</Text>
+              <Text style={styles.taskStatus}>{task.completed ? 'Completed' : 'Incomplete'}</Text>
+              <Rating
+                style={styles.rating}
+                imageSize={30}
+                ratingCount={5}
+                startingValue={rating}
+                readonly={true}
+              />
+            </View>
+            <View style={styles.actionButtons}>
+              <IconButton icon="pencil" onPress={() => setIsEditing(true)} />
+              <IconButton icon="delete" onPress={() => onDelete(task.name)} />
+            </View>
+          </View>
+      </>
+  ) : (
```

Comment:
At what point do we set isEditing back to false once we are done editing?

Reply:
Out of scope for this PR T_T